torrent/torrent.go

```diff
@@ -71,24 +105,29 @@ func (b *TorrentWorker) start() {
 			switch command.(type) {
 			case *FetchCommand:
 
-				cmd := command.(*FetchCommand)
-				if lc := b.getLaunchContext(cmd.LaunchContext); lc == nil {
-					glog.Errorf("Incoming event was not a known launch context: %T", cmd.LaunchContext)
+				authAttribs, err := authAttributes(b.db)
+				if err != nil {
+					glog.Error(err)
 				} else {
 
-					glog.V(2).Infof("URL to fetch: %s\n", lc.URL())
+					cmd := command.(*FetchCommand)
+					if lc := b.getLaunchContext(cmd.LaunchContext); lc == nil {
+						glog.Errorf("Incoming event was not a known launch context: %T", cmd.LaunchContext)
+					} else {
+						glog.V(2).Infof("URL to fetch: %s\n", lc.URL())
 
-					// TODO: decide where the best place is to shortcut the fetch call if the docker images it names are already in the local repo
-					// (could be here or bypass this worker altogether)
-					// (this is really important because we want to be able to delete the downloaded image files after docker load)
+						// TODO: decide where the best place is to shortcut the fetch call if the docker images it names are already in the local repo
+						// (could be here or bypass this worker altogether)
+						// (this is really important because we want to be able to delete the downloaded image files after docker load)
 
-					imageFiles, err := fetch.PkgFetch(b.Config.Collaborators.HTTPClientFactory.WrappedNewHTTPClient(), lc.URL(), lc.Signature(), b.Config.Edge.TorrentDir, b.Config.Edge.CACertsPath, b.Config.UserPublicKeyPath())
+						imageFiles, err := fetch.PkgFetch(b.Config.Collaborators.HTTPClientFactory.WrappedNewHTTPClient(), lc.URL(), lc.Signature(), b.Config.Edge.TorrentDir, b.Config.Edge.CACertsPath, b.Config.UserPublicKeyPath(), authAttribs)
```

Comment:
This is the updated call; note that the ..Edge.CACertsPath is in the args per that bug fix.

Reply:
The lib is here: https://github.com/open-horizon/horizon-pkg-fetch/blob/master/fetch.go#L368;